packages/storage-types/index.d.ts

```diff
@@ -85,24 +84,41 @@ export interface UploadMetadata extends SettableMetadata {
   md5Hash?: string | null;
 }
 
+export interface FirebaseStorageError {
+  name: string;
+  code: string;
+  message: string;
+  serverResponse: null | string;
+}
+
+export type NextFn<T> = (value: T) => void;
+export type ErrorFn = (error: FirebaseStorageError) => void;
+export type CompleteFn = () => void;
+export type Unsubscribe = () => void;
+export interface StorageObserver<T> {
+  next?: NextFn<T> | null;
+  error?: ErrorFn | null;
+  complete?: CompleteFn | null;
+}
+
 export interface UploadTask {
   cancel(): boolean;
-  catch(onRejected: (a: Error) => any): Promise<any>;
+  catch(onRejected: (a: FirebaseStorageError) => any): Promise<any>;
   on(
     event: TaskEvent,
     nextOrObserver?:
-      | Partial<Observer<UploadTaskSnapshot>>
+      | Partial<StorageObserver<UploadTaskSnapshot>>
       | null
-      | ((a: UploadTaskSnapshot) => any),
-    error?: ((a: Error) => any) | null,
+      | ((a: UploadTaskSnapshot) => unknown),
```

Comment:
This `unknown` seems inconsistent with the other return types.

Since this is going to land in a breaking change release, do you think it makes sense to start using more descriptive names than "a"? (here and everywhere)

Reply:
Sure, seems like a good idea.